### PR TITLE
net/aodvv2: refactor RCS code

### PR DESCRIPTION
**Pull request by jeandudey**
- Refactors the code of RCS to make it testable.
- Adds tests for RCS.
- Improves documentation.

-----

- I'll add a GitHub Action to compile this.

### CODE REVIEW

File: tests/aodvv2_rcs/main.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test AODVv2 Router Client Set
+ *
+ * @author      Jean Pierre Dudey <jeandudey@hotmail.com>
```
Comment (luisan00) [inline tests/aodvv2_rcs/main.c:16]: use the same header format that is being used  in the project

File: tests/aodvv2_rcs/main.c [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#include "embUnit.h"
+#include "embUnit/embUnit.h"
+
+#include "net/aodvv2/conf.h"
+#include "net/aodvv2/rcs.h"
+
+#define TEST_ASSERT_EQUAL_ADDR(a, b) \
+    TEST_ASSERT(ipv6_addr_equal((a), (b)))
+
+static void _set_up(void)
```
Comment (luisan00) [inline tests/aodvv2_rcs/main.c:30]: This function is for...?

Comment (jeandudey) [reply tests/aodvv2_rcs/main.c:30]: Placeholder, needs to be there.

**Comment by jeandudey** [comment]
@xkevin190 could you give me a review? :wink:

**Comment by jeandudey** [comment]
@xkevin190 ping :wink:

**Pull request merged**